mnefun/_report.py

```diff
@@ -251,6 +252,36 @@ def _report_raw_psd(report, raw, raw_pca=None, p=None):
     print('%5.1f sec' % ((time.time() - t0),))
 
 
+def _peak_times(evoked, max_peaks=5):
+    '''Return times of prominent peaks from global field power.'''
+    gfp_list = []
+    if 'meg' in evoked:
+        evk = evoked.copy().pick_types(meg=True)
+        gfp = evk.data.std(axis=0)
+        gfp_list.append(gfp / gfp.max())
+    if 'eeg' in evoked:
+        evk = evoked.copy().pick_types(meg=False, eeg=True)
+        gfp = evk.data.std(axis=0)
+        gfp_list.append(gfp / gfp.max())
+    if not gfp_list:    # for non-meg/eeg data types
+        gfp = evoked.data.std(axis=0)
+        gfp = gfp / gfp.max()
+    else:
+        gfp = np.array(gfp_list).mean(axis=0)
```

Comment:
I think the negative peak times is a result of the "prominence" algorithm, which can skip minor peaks too close to a major one; thus spurious energy at negative times makes the top-5 peaks.  This would likely occur in any scenario where negative times aren't suppressed.  One way around it is to limit the number of peaks (e.g. 3 instead of the current max of 5).  As I wrote before though, at least for the report negative peaks may not be so bad as they provide indication of activity during the baseline period.

If we do definitely want to avoid negative times, without explicitly disallowing them, whitening seems like a good idea.  The example you show above is compelling, as the baseline variance is quite low.  Will there be difficulties when an evoked object doesn't represent continuous time, but peaks are still meaningful for a report (e.g. a spectrum)?

Reply:
> Will there be difficulties when an evoked object doesn't represent continuous time, but peaks are still meaningful for a report (e.g. a spectrum)?

The whitening is spatial so if the spectrum linearity is preserved (i.e., kept complex / not taking the magnitude) it would still be fine. But I think we're a ways away from doing anything like this in the built-in report generation, it seems better suited to custom paradigm-specific code.

Want to take a stab at the whitening to see if it makes things work better?

Reply:
Yes, I'll be happy to try a whitening approach for time-series peak detection.  It will take a bit of time to make sure I understand all the steps - I'll start as you suggest with the whitening section of the report.

On a side note I was thinking that it would be too restrictive to simply prohibit detection of negative-time peaks.  I can imagine an experimental paradigm for which signal energy prior to some 0-time reference might be expected (e.g. a sequence of event types, like a saccade or response to an auditory cue preceding a flashed visual target).

Reply:
Ah, and now I see why you said the whitening also normalizes across sensor types: From the figure, the baseline power is 1.0 on average (red dotted line).

Reply:
Agreed it's better not to force non-negative